### PR TITLE
EWPP-2832: Add Iframe title to media types using the Iframe source.

### PR DESCRIPTION
**Pull request by 22Alexandra**

### CODE REVIEW

File: modules/oe_media_iframe/config/install/field.field.media.iframe.oe_media_iframe_title.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  config:
+    - field.storage.media.oe_media_iframe_title
+    - media.type.iframe
+id: media.iframe.oe_media_iframe_title
+field_name: oe_media_iframe_title
+entity_type: media
+bundle: iframe
+label: 'Iframe title'
+description: 'Providing an Iframe title value will replace the title value in the iframe html.'
+required: false
```
Comment (nagyad) [inline modules/oe_media_iframe/config/install/field.field.media.iframe.oe_media_iframe_title.yml:13]: Should be required.

File: modules/oe_media_iframe/config/install/field.field.media.video_iframe.oe_media_iframe_title.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  config:
+    - field.storage.media.oe_media_iframe_title
+    - media.type.video_iframe
+id: media.video_iframe.oe_media_iframe_title
+field_name: oe_media_iframe_title
+entity_type: media
+bundle: video_iframe
+label: 'Iframe title'
+description: 'Providing an Iframe title value will replace the title value in the iframe html.'
+required: false
```
Comment (nagyad) [inline modules/oe_media_iframe/config/install/field.field.media.video_iframe.oe_media_iframe_title.yml:13]: Should be required.

File: modules/oe_media_iframe/oe_media_iframe.post_update.php [hunk 3]
```diff
@@ -176,3 +178,65 @@ function oe_media_iframe_post_update_00006(): void {
...
+
+  foreach ($iframe_types as $type) {
+    $id = $type->id();
+    $field_config = FieldConfig::load("media.$id.oe_media_iframe_title");
+    if ($field_config) {
+      // If the current media type already has the field, we skip it.
+      continue;
+    }
+    $field = $entity_type_manager
+      ->getStorage('field_config')
```
Comment (nagyad) [inline modules/oe_media_iframe/oe_media_iframe.post_update.php:213]: Make a field config storage variable outside of the loop.

File: modules/oe_media_iframe/oe_media_iframe.post_update.php [hunk 4]
```diff
@@ -176,3 +178,65 @@ function oe_media_iframe_post_update_00006(): void {
...
+      ->create([
+        'field_storage' => $storage,
+        'bundle' => $type->id(),
+        'label' => 'Iframe title',
+        'description' => 'Providing an Iframe title value will replace the title value in the iframe html.',
```
Comment (nagyad) [inline modules/oe_media_iframe/oe_media_iframe.post_update.php:218]: Provides an Iframe title value which will replace the title value in the iframe html.

File: modules/oe_media_iframe/oe_media_iframe.post_update.php [hunk 5]
```diff
@@ -176,3 +178,65 @@ function oe_media_iframe_post_update_00006(): void {
...
+        'required' => FALSE,
```
Comment (nagyad) [inline modules/oe_media_iframe/oe_media_iframe.post_update.php:219]: Should be required.

File: modules/oe_media_iframe/src/Plugin/media/Source/Iframe.php [hunk 6]
```diff
@@ -94,6 +94,31 @@ public function createSourceField(MediaTypeInterface $type) {
...
+      $storage = $fields['oe_media_iframe_title'];
+    }
+    /** @var \Drupal\field\FieldConfigInterface $field */
+    $field = $this->entityTypeManager
+      ->getStorage('field_config')
+      ->create([
+        'field_storage' => $storage,
+        'bundle' => $type->id(),
+        'label' => 'Iframe title',
+        'description' => 'Providing an Iframe title value will replace the title value in the iframe html.',
```
Comment (nagyad) [inline modules/oe_media_iframe/src/Plugin/media/Source/Iframe.php:117]: Providing an Iframe title value will replace the title value in the iframe html.

File: modules/oe_media_iframe/src/Plugin/media/Source/Iframe.php [hunk 7]
```diff
@@ -94,6 +94,31 @@ public function createSourceField(MediaTypeInterface $type) {
...
+        'required' => FALSE,
```
Comment (nagyad) [inline modules/oe_media_iframe/src/Plugin/media/Source/Iframe.php:118]: Should be required.

**Review by nagyad — CHANGES_REQUESTED** [review]

File: modules/oe_media_iframe/src/Plugin/Field/FieldFormatter/MediaIframeFormatter.php [hunk 8]
```diff
@@ -108,10 +108,23 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
     $text_format = $media_type->getSource()->getConfiguration()['text_format'] ?? NULL;
 
     foreach ($items as $delta => $item) {
+      $media = $item->getEntity();
```
Comment (sergepavle) [inline modules/oe_media_iframe/src/Plugin/Field/FieldFormatter/MediaIframeFormatter.php:111]: Is it intentional that it is done not within oe_theme or in some additional format filter?

**Review by sergepavle — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Pull request merged**